### ISSUE TITLE
CI: daily workflow testing more versions in back-compat

### ISSUE THREAD

**Issue by bradleystachurski** [fedimint/fedimint#5180]
Now that we're using self-hosted CI runners, we should create a daily workflow that runs backwards-compatibility tests against more versions.

CI for PRs only test compatibility vs the oldest supported version ([v0.2.1](https://github.com/fedimint/fedimint/blob/276785214288cdc5b0202aed0989fa0ff25ef18d/.github/workflows/ci-backwards-compatibility.yml#L57) as-of creating this issue). We originally decided to run against fewer versions to improve developer experience for PRs, but this makes it possible for backwards-incompatible changes to go undetected (see: https://github.com/fedimint/fedimint/pull/5171).

We should also consider other tests that don't fit within PR CI workflows but would be useful to run in daily workflows, e.g. upgrade tests, but that's outside the scope of this issue.

Other considerations:
- Use the [full matrix](https://github.com/fedimint/fedimint/blob/773ff57051aa0b38e5b89699ce8b92e58090ee03/scripts/_common.sh#L233) instead of the partial
  - Might take too long against multiple versions, so we may still use the [partial matrix](https://github.com/fedimint/fedimint/blob/773ff57051aa0b38e5b89699ce8b92e58090ee03/scripts/_common.sh#L265)
- Run the daily cron during low volume hours (midnight PT?)

**Comment by elsirion** [comment]
Yeah, I think we could make the hourly check more extensive but reduce frequency, possibly to daily or a few times around low demand hours.
